packages/agents/lighthouse/src/tasks/propose/propose.ts

```diff
@@ -140,14 +154,17 @@ export const makePropose = async (config: NxtpLighthouseConfig, chainData: Map<s
             jsonifyError(e as NxtpError),
             { spokeDomain },
           );
+
+          proposeSpokeSuccess = false;
         }
       }
     } else if (rootManagerMode.mode === ModeType.SlowMode) {
       context.logger.info("In Slow Mode. No op.", requestContext, methodContext);
     } else {
       throw new Error(`Unknown mode detected: ${JSON.stringify(rootManagerMode)}`);
     }
-    if (context.config.healthUrls.propose) {
+
+    if (context.config.healthUrls.propose && proposeHubSuccess && proposeSpokeSuccess) {
```

Comment:
I am still wondering why it shouldn't send a heartbeat if proposing the snapshot on one of the spokes fails.

Reply:
so were alerted if messaging halts

Reply:
How will we know of that failure ?
We could increase the heartbeat threshold to allow for multiple rounds of proposal failures.